netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,14 +79,18 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
```

Comment:
IMHO we were broken when we only had `forAddress`. `InetSocketAddress`, being able to resolve a name in itself, is not the correct way to pass a name that is to be resolved somewhere else. If we just use the host name and port from `InetSocketAddress` and discard the resolved address, we are changing the semantics of `InetSocketAddress`. The current semantics of `InetSocketAddress` is unfortunately problematic, but I don't think we are in the position to change it.

`forTarget` is the canonical way, and I would prefer it to be the only way, to pass a name that is to be resolved by `NameResolver`s.

If you are worried about users shooting themselves in the foot, we could even log a warning in `forAddress` when it receives an `InetSocketAddress`.

Looks we cannot convince each other. Maybe we should hear other people's opinions.
@ejona86 @louiscryan @carl-mastrangelo @madongfly